src/stlink-lib/usb.c

```diff
@@ -1273,6 +1273,7 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, enum connect_type connect,
 
     if (cnt < 0) {
         WLOG ("Couldn't find %s ST-Link devices\n", (devBus && devAddr) ? "matched" : "any");
+        libusb_free_device_list(list, 1);
```

Comment:
If we want to use `libusb_free_device_list` on error, then we must force initialize the `list` variable before calling `libusb_get_device_list`.
As I understand it, the bug is a Windows backend. Better to wrap this call in `#ifdef _WIN32 ... #endif` and add a comment that this is a bug fix.

Reply:
Hi @Ant-ON , I am not sure if it is a windows-only bug, from what I understood the correct procedure is to always free the devices (like we do on line 1287, for example)

And about making sure that the list was initialized: it must have been initialized at line 1209, the only way of arriving here without having the list initialized is if the call to libusb_get_device_list at line 1209 returned an error, and in this case we would already have problems starting from the while loop at line 1235, which is not accounting for a negative cnt value coming from the error code,  in my opinion, this is the actual issue we must address

Reply:
Hi @Ant-ON  , I am not sure if it is a windows-only bug, from what I understood the correct procedure is to always free the devices (like we do on line 1287, for example)

And I initialized the list, so it is not a problem if we try to free a list that was not allocated (it would just return [here](https://github.com/libusb/libusb/blob/e782eeb2514266f6738e242cdcb18e3ae1ed06fa/libusb/core.c#L879))

I also added a check for cnt > 0, as libusb_get_device_list (line 1209) could return a negative value in case of an error, leading to out of bounds memory access inside the while loop

Reply:
@gustavolaureano I agree, now there should be no errors in the work. Thank you!